### ISSUE TITLE
Maplike errors is not fixed (mojang issue?)

### ISSUE THREAD

**Issue by broken1arrow** [PaperMC/Paper#4236]
### What behaviour is expected:
Not console errors :)

### What behaviour is observed:
[20:25:33] [Paper Async Chunk Task Thread #0/ERROR]: No key position_predicate in MapLike

### Steps/models to reproduce:
Start a 16.1 world and upgrade to 1.16.2 (lates was 148 when i try)

### Paper version:
It was 16.1 papermc 133 to 16.2 papermc 148
### Anything else:
https://pastebin.com/shetqD5e

**Comment by electronicboy** [comment]
The issue toasting villagers was fixed, MapLike issues in general are still expected, but, the major one is fixed

**Comment by broken1arrow** [comment]
> 
> 
> The issue toasting villagers was fixed, MapLike issues in general are still expected, but, the major one is fixed

oright, thank you for the explanation.

**Comment by aikar** [comment]
Going to reopen as want to see about this MapLike issue, as this is a different one than the other.

Interestingly, it's back to the original 1.16.1 error message, but not triggered the same way.

**Comment by broken1arrow** [comment]
> 
> 
> Going to reopen as want to see about this MapLike issue, as this is a different one than the other.
> 
> Interestingly, it's back to the original 1.16.1 error message, but not triggered the same way.

Is it very strange bug. I don´t think mojang fix it, if you not report it on vanilla server.
So probably goo unfixed.

**Comment by aikar** [comment]
I added your log to the mojang ticket.

Do you have any context of what triggered it? do you have a backup of the original chunk?

**Comment by MJRLegends** [comment]
Yea this looks a lot like the old MC 1.16/1.16.1 error said to be fixed in MC 1.16.2 https://bugs.mojang.com/browse/MC-191388 probably means part of the issue might of not got fixed by mojang

**Comment by broken1arrow** [comment]
> 
> 
> I added your log to the mojang ticket.
> 
> Do you have any context of what triggered it? do you have a backup of the original chunk?

No, ofc i have backups but I not upgrade on the main server so, this server use a older version (not super old it has 1.16.1 upgrade at least). 

I can sen an backup, but are not same has first report.

I try it in single player from 1.16.1 to 1.16.2 and yes chunks is preloaded in older version (1.16.1 last time). Cord is -3218 80 -557
https://pastebin.com/2KGSSDqH

If fix the world in 16.1, so you update and login (i think you end up in right spot).

**Comment by broken1arrow** [comment]
So take some time to upload ;) https://www.dropbox.com/s/rgv0bwxsedf77kw/world%201.16.1.zip?dl=0 I hope not any exploit the seed (but what to expect from internet ether way) :)

Yes i read in that thread on mojang :)

**Comment by broken1arrow** [comment]
I start a issue https://bugs.mojang.com/browse/MC-199174

But the mod say is same problem as yours  aikar (they say soo often) :)

**Comment by broken1arrow** [comment]
Only error i get no plugins is it.the file i send here i use now. I don't have the time to test it on vanilla, i can try it later. I use papermc 148
cord 1531 80 -4985
`[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[10:54:57] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree`

628 80 -5018
`[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:taiga_grass
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:taiga_grass
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:taiga_grass
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:pine_tree
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:taiga_grass
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:pine_tree
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:taiga_grass
[11:02:49] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:pine_tree
`

1636 80 1789
`[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:hay_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:hay_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:melon_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:melon_pile
`

1854 80 1394
`[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:hay_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:hay_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:melon_pile
[11:39:19] [Paper Async Chunk Task Thread #2/ERROR]: Unknown registry key: minecraft:melon_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:cactus
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:cactus
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[11:44:55] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:cactus`

**Comment by broken1arrow** [comment]
So have time to test in vanilla, so is something with your fix (but i don't notes any worlds corruptions). It seams like it already loaded chunks some get the problem. i don't know if the download link work, so put up different uploaded https://anonfiles.com/X4k3F7Rdo8/world_1.16.1_zip
cords 1531.5, 80.0, -4984.5
```
10:30:23.562
[CHAT] Teleported broken_arrow1 to 1531.5, 80.0, -4984.5
10:30:25.480
Input does not contain a key [type]: MapLike[{name:"minecraft:plain_flower",config:{}}]
10:30:25.486
Input does not contain a key [type]: MapLike[{name:"minecraft:normal_tree",config:{}}]
10:30:25.497
Input does not contain a key [type]: MapLike[{name:"minecraft:plain_flower",config:{}}]
10:30:25.501
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:30:25.507
Input does not contain a key [type]: MapLike[{name:"minecraft:plain_flower",config:{}}]
10:30:25.507
Input does not contain a key [type]: MapLike[{name:"minecraft:normal_tree",config:{}}]
10:30:25.507
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:30:25.507
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:30:25.508
Input does not contain a key [type]: MapLike[{name:"minecraft:normal_tree",config:{}}]
```

more here. 
```
[CHAT] Teleported broken_arrow1 to 628.5, 80.0, -5017.5
10:34:44.496
Input does not contain a key [type]: MapLike[{name:"minecraft:taiga_grass",config:{}}]
10:34:44.496
Input does not contain a key [type]: MapLike[{name:"minecraft:taiga_grass",config:{}}]
10:34:44.504
Input does not contain a key [type]: MapLike[{name:"minecraft:taiga_grass",config:{}}]
10:34:44.505
Input does not contain a key [type]: MapLike[{name:"minecraft:pine_tree",config:{}}]
10:34:44.506
Input does not contain a key [type]: MapLike[{name:"minecraft:taiga_grass",config:{}}]
10:34:44.507
Input does not contain a key [type]: MapLike[{name:"minecraft:pine_tree",config:{}}]
10:34:44.508
Input does not contain a key [type]: MapLike[{name:"minecraft:taiga_grass",config:{}}]
10:34:44.509
Input does not contain a key [type]: MapLike[{name:"minecraft:pine_tree",config:{}}]
```
```
10:36:25.539
[CHAT] Teleported broken_arrow1 to 1636.5, 80.0, 1789.5
10:36:25.887
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:36:25.890
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:36:25.893
Input does not contain a key [type]: MapLike[{name:"minecraft:melon_pile",config:{}}]
10:36:25.893
Input does not contain a key [type]: MapLike[{name:"minecraft:melon_pile",config:{}}]
```
```
10:37:22.867
[CHAT] Teleported broken_arrow1 to 1854.5, 80.0, 1394.5
10:37:25.129
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.130
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.130
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.131
Input does not contain a key [type]: MapLike[{name:"minecraft:cactus",config:{}}]
10:37:25.132
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.134
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.134
Input does not contain a key [type]: MapLike[{name:"minecraft:cactus",config:{}}]
10:37:25.135
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.137
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.138
Input does not contain a key [type]: MapLike[{name:"minecraft:hay_pile",config:{}}]
10:37:25.138
Input does not contain a key [type]: MapLike[{name:"minecraft:cactus",config:{}}]

```

**Comment by broken1arrow** [comment]
I try papermc 184 with same map i put higher up.
```

[12:53:22] [Server thread/INFO]: broken_arrow1 issued server command: /tp 1531.5 80.0 -4984.5
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:plain_flower
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:53:24] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:normal_tree

[12:55:08] [Server thread/INFO]: broken_arrow1 issued server command: /tp 1636.5 80.0 1789.5
[12:55:09] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[12:55:09] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:hay_pile
[12:55:09] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:melon_pile
[12:55:09] [Paper Async Chunk Task Thread #1/ERROR]: Unknown registry key: minecraft:melon_pile

[12:56:01] [Server thread/INFO]: broken_arrow1 issued server command: /tp 1854.5 80.0 1394.5
[12:56:01] [Server thread/WARN]: broken_arrow1 moved too quickly! 225.12227839081515,-1.1194792381865852,-398.6119979592545
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:cactus
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:cactus
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:hay_pile
[12:56:02] [Paper Async Chunk Task Thread #0/ERROR]: Unknown registry key: minecraft:cactus

[12:57:31] [Server thread/INFO]: broken_arrow1 issued server command: /tp 628.5 80.0 -5017.5
[12:57:31] [Server thread/WARN]: broken_arrow1 moved too quickly! -1213.8671934642841,-16.309625136286385,-6291.938099901076
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:taiga_grass
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:taiga_grass
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:taiga_grass
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:pine_tree
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:taiga_grass
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:pine_tree
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:taiga_grass
[12:57:32] [Paper Async Chunk Urgent Task Thread/ERROR]: Unknown registry key: minecraft:pine_tree
```

**Comment by broken1arrow** [comment]
some images, i dont see any bigger issues with villgers (only some random blocks on pathway and more roads an house in other spots).
![2020-09-05_12 54 50](https://user-images.githubusercontent.com/16368286/92303746-255a2480-ef78-11ea-8f87-cad602532d64.png)
![2020-09-05_12 58 26](https://user-images.githubusercontent.com/16368286/92303749-268b5180-ef78-11ea-897d-3d94f9c79600.png)
![2020-09-05_12 54 40](https://user-images.githubusercontent.com/16368286/92303751-27bc7e80-ef78-11ea-850f-7c06cfd70c4c.png)

**Comment by electronicboy** [comment]
> Unknown registry key

That's normal as is stated on discord, that's due to world gen features which mojang removed from the game, but didn't remove from existing data; those specifically, at least the ones we've seen, are expected until mojang starts cleaning stuff up...

**Comment by broken1arrow** [comment]
> 
> 
> > Unknown registry key
> 
> That's normal as is stated on discord, that's due to world gen features which mojang removed from the game, but didn't remove from existing data; those specifically, at least the ones we've seen, are expected until mojang starts cleaning stuff up...

I´m has spend some time around this, because of what @aikar say. So I working hard to pinpoint and replicate error and put up world file and this was for nothing?

**Comment by mowriz** [comment]
after upgrade from 1.16.3 to 1.16.4 its still an issue.

`[20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree
[20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree
[20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree
[20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree
[20:49:51 ERROR]: Unknown registry key: minecraft:normal_tree
[20:50:17 ERROR]: Unknown registry key: minecraft:normal_tree
[20:50:17 ERROR]: Unknown registry key: minecraft:normal_tree`

**Comment by broken1arrow** [comment]
> after upgrade from 1.16.3 to 1.16.4 its still an issue.
> 
> `[20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree [20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree [20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree [20:45:35 ERROR]: Unknown registry key: minecraft:normal_tree [20:49:51 ERROR]: Unknown registry key: minecraft:normal_tree [20:50:17 ERROR]: Unknown registry key: minecraft:normal_tree [20:50:17 ERROR]: Unknown registry key: minecraft:normal_tree`

If you have old chunks som not get loaded in 1.16.3, you get same problem in 1.16.4.

If they not fix it in 1.17 you still can get this problem.
